### ISSUE TITLE
Add option to use an API keys for users that don't have a password

### ISSUE THREAD

**Issue by mcfisch** [badjware/certbot-dns-cpanel#12]
I'm trying to configure an automated setup for all my little things that I have behind several reverse proxies and therefore need a wildcard certificate to make the whole process less cumbersome. The big problem: my DNS provider only allows logging into cPanel through their own webhosting portal - and that works without actually giving me the password for the cPanel user. This is a problem if you want to use the plugin that relies on full user login credentials.

However, I've found a way to address the cPanel UI by using an API key that I created with said cPanel user, and that works when running requests manually against the API, such as:
```
curl -H'Authorization: cpanel ${my_user}:${my_api_key}' 'https://${my_domain}:2083/execute/DNS/lookup?domain=${my_domain}'
```

When debugging the `dns_cpanel.py` script to find out how to adapt it I noticed it is using `basic_auth` with base64 encoding. As a simple workaround for myself I've replaced the lines in the original script (https://github.com/badjware/certbot-dns-cpanel/blob/master/certbot_dns_cpanel/dns_cpanel.py#L79-L83)
```
        self.headers = {
            'Authorization': 'Basic %s' % base64.b64encode(
                ("%s:%s" % (username, password)).encode()
            ).decode('utf8')
        }
```
with this:
```
        self.headers = {
            'Authorization': 'cpanel %s:%s' % (username, password)
        }
```
This works well with the JSON api the script using. Maybe something like this is worth to be included, or if this is too much effort to implement in a more reliable and secure way (though since these configs are plain text there isn't much to secure anyway...).

Let me know if you'd like to work on this on your end or if it makes more sense to run a seperate plugin for this.

**Comment by mcfisch** [comment]
And after creating the issue here I've noticed that somebody actually created a PR with a more sophisticated way to address this... 🙄 😉  ... 
#10

**Comment by 0x3333** [comment]
I also created a version with an option to install in a cPanel server. https://github.com/0x3333/certbot-cpanel

**Comment by badjware** [comment]
This issue has been fixed with #10 being included in version 0.3.0.
